databallpy/data_parsers/event_data_parsers/statsbomb_parser.py

```diff
@@ -0,0 +1,507 @@
+        raise FileNotFoundError(f"File {events_loc} does not exist.")
+    
+    if not os.path.exists(match_loc):
+        LOGGER.error(f"File {match_loc} does not exist.")
+        raise FileNotFoundError(f"File {match_loc} does not exist.")
+    
+    if not os.path.exists(lineup_loc):
+        LOGGER.error(f"File {lineup_loc} does not exist.")
+        raise FileNotFoundError(f"File {lineup_loc} does not exist.")
+
+    if not isinstance(pitch_dimensions, (tuple, list)) or len(pitch_dimensions) != 2:
+        LOGGER.error(
+            f"Invalid pitch_dimensions: {pitch_dimensions}. "
+            "Must be a tuple of length 2."
+        )
+        raise ValueError(
+            f"Invalid pitch_dimensions: {pitch_dimensions}. "
+            "Must be a tuple of length 2."
+        )
+    
+    # Load the metadata
+    metadata = _load_metadata(match_loc, lineup_loc, pitch_dimensions)
+    LOGGER.info("Successfully loaded Statsbomb metadata.")
+    # Load the event data
+    event_data, databallpy_events, metadata = _load_event_data(events_loc, metadata, pitch_dimensions)
+    
+    LOGGER.info("Successfully loaded Statsbomb event data and databallpy events.")
+    return event_data, metadata, databallpy_events
+
+def _load_metadata(match_loc: str, lineup_loc: str, pitch_dimensions: tuple) -> Metadata:
+    """Function to load metadata from the match.json and lineup.json files
+
+    Args:
+        match_loc (str): location of the match.json file
+        lineup_loc (str): location of the lineup.json file
+        pitch_dimensions (tuple): the length and width of the pitch in meters
+
+    Returns:
+        MetaData: all metadata information of the current match
+    """
+    with open(match_loc, "r", encoding="utf-8") as f:
+        match_json = json.load(f)[0]
+
+    with open(lineup_loc, "r", encoding="utf-8") as f:
+        lineup_json= json.load(f)
+    
+    home_index = 0 if lineup_json[0]["team_id"] == match_json["home_team"]["home_team_id"] else 1
+    away_index = 1 if home_index == 0 else 0
+
+    home_players = _get_player_info(lineup_json[home_index]["lineup"])
+    away_players = _get_player_info(lineup_json[away_index]["lineup"])
+
+    periods = {
+        "period_id": [1, 2, 3, 4, 5],
+        "start_datetime_ed": [],
+        "end_datetime_ed": [],
+    }
+    match_start = pd.to_datetime(match_json["match_date"]+" "+match_json["kick_off"], utc=True)
+    for id in range(5):
+        if id == 0:
+            periods["start_datetime_ed"].append(match_start)
+            periods["end_datetime_ed"].append(match_start + pd.to_timedelta(45, unit="minutes"))
+        elif id == 1:
+            periods["start_datetime_ed"].append(match_start+ pd.to_timedelta(60, unit="minutes"))
+            periods["end_datetime_ed"].append(match_start + pd.to_timedelta(105, unit="minutes"))
+        else:
+            periods["start_datetime_ed"].append(pd.NaT)
+            periods["end_datetime_ed"].append(pd.NaT)
```

Comment:
Not strictly necisary, but append is a slow operation, and new variables are saved which are uses once, which is generally only done if it increases readbality greatly, but I think that is not the case here.

```suggestion
 match_start = pd.to_datetime(match_json["match_date"]+" "+match_json["kick_off"], utc=True)
 periods = {
        "period_id": [1, 2, 3, 4, 5],
        "start_datetime_ed": [
              match_start, 
              match_start + pd.to_timedelta(60, unit="minutes"),  
              pd.NaT,  
              pd.NaT, 
              pd.NaT 
        ],
        "end_datetime_ed": [ 
              match_start + pd.to_timedelta(45, unit="minutes")
              match_start + pd.to_timedelta(105, unit="minutes"),  
              pd.NaT,  
              pd.NaT, 
              pd.NaT 
        ],
    }
```